docs/google-test.md

```diff
@@ -0,0 +1,278 @@
+- name: The target name, aka the name of the binary to build
+- srcs: The source file containing the tests
+- deps: The dependencies of the build; here, `//sdk/src/trace` refers to the directory of the source code being tested, and `@com_google_googletest//:gtest_main` is the Google Test dependency.
+
+
+Building and executing a unit test is also very simple. On terminal, navigate to the directory the WORKSPACE file resides in, and execute two commands:
+
+```
+bazel build //path/to/package:target-name
+bazel-bin/path/to/package/target-name
+```
+The first command builds the unit test with the given target name; the second runs the unit test.
+
+This next command runs the above two commands at once, though you may get a message stating `There were tests whose specified size is too big` and they may not run without modifying some options.
+
+```
+bazel test //path/to/package:target-name
+```
+
+For our example the commands above would be:
+```
+bazel build //sdk/test/trace:tracer_provider_test
+bazel-bin/sdk/test/trace/tracer_provider_test
+```
+or
+```
+bazel test //sdk/test/trace:tracer_provider_test
+```
+
+Additionally, all Bazel tests can be ran in a Docker container by navigating to the root of the directory and executing the command:
+```
+./ci/run_docker.sh ./ci/do_ci.sh bazel.test
+```
+
+### CMake
+
+CMake can be installed following instructions from [here](https://cmake.org/download/). You will also need to install Google Test, along with libbenchmark.
+
+Download a zip of the Google Test source code [here](https://github.com/google/googletest/releases/tag/release-1.10.0). Then, navigate into the source code directory and run the following commands:
+
+```
+mkdir build && cd build
+cmake ..
+make
+make install
+```
+
+This will install Google Test for you; you can delete the source code after running these commands.
+
+Next, follow the instructions [here](https://github.com/google/benchmark#installation) to download Google Benchmark. The section following this one will detail performance benchmarking using Google Benchmark.
+
+
+CMake makes use of a `CMakeLists.txt` file that provides instructions for a project’s source files and targets. There is a main `CMakeLists.txt` file in the root of the project directory, and a `CMakeLists.txt` for each subdirectory. In the main `CMakeLists.txt` file, the `cmake_minimum_required`  and `CMAKE_CXX_STANDARD` are set, the project name is set with `project()`, directories are included and subdirectories added, and much more. However, more importantly, in the context of Google Test within the OTel C++ repo, the `CMakeLists.txt` to look at is the one located in the directory your tests are being written in; for this example, it’s `opentelemetry-cpp/sdk/test/trace`. This [CMakeLists.txt](https://github.com/open-telemetry/opentelemetry-cpp/blob/master/sdk/test/trace/CMakeLists.txt) file, shown below, describes the names of the tests to be added as executables.
```

Comment:
seems to have non-ASCII characters here?

Reply:
Not sure where exactly these non-ASCII characters are here